src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Payloads;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using Microsoft.VisualStudio.TestPlatform.VsTestConsole.TranslationLayer.Interfaces;
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine;
+
+/// <summary>
+/// The in-process wrapper.
+/// </summary>
+internal class InProcessVsTestConsoleWrapper : IVsTestConsoleWrapper
+{
+    // Must be in sync with the highest supported version in
+    // src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs file.
+    private readonly int _highestSupportedVersion = 6;
+
+    private readonly ITranslationLayerRequestSender _requestSender;
+    private readonly ITestPlatformEventSource _testPlatformEventSource;
+    private bool _isInitialized;
+    private bool _sessionStarted;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="InProcessVsTestConsoleWrapper"/>.
+    /// </summary>
+    /// 
+    /// <param name="consoleParameters">The console parameters.</param>
+    public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
+        : this(
+              consoleParameters,
+              requestSender: new VsTestConsoleRequestSender(),
+              testRequestManager: null,
+              executor: new Executor(ConsoleOutput.Instance),
+              testPlatformEventSource: TestPlatformEventSource.Instance)
+    { }
+
+    internal InProcessVsTestConsoleWrapper(
+        ConsoleParameters consoleParameters,
+        ITranslationLayerRequestSender requestSender,
+        ITestRequestManager? testRequestManager,
+        Executor executor,
+        ITestPlatformEventSource testPlatformEventSource)
+    {
+        EqtTrace.Info("VsTestConsoleWrapper.StartSession: Starting VsTestConsoleWrapper session.");
+
+        _testPlatformEventSource = testPlatformEventSource;
+        _testPlatformEventSource.TranslationLayerInitializeStart();
+
+        // Start communication.
+        _requestSender = requestSender;
+        var port = _requestSender.InitializeCommunication();
+        if (port <= 0)
+        {
+            // Close the sender as it failed to host server.
+            _requestSender.Close();
+            throw new TransationLayerException("Error hosting communication channel.");
+        }
+
+        // Fill the parameters.
+        consoleParameters.ParentProcessId = Process.GetCurrentProcess().Id;
+        consoleParameters.PortNumber = port;
+
+        // Start vstest.console.
+        // TODO: under VS we use consoleParameters.InheritEnvironmentVariables, we take that
+        // into account when starting a testhost, or clean up in the service host, and use the
+        // desired set, so all children can inherit it.
+        foreach (var pair in consoleParameters.EnvironmentVariables)
+        {
+            Environment.SetEnvironmentVariable(pair.Key, pair.Value);
```

Comment:
Can we use the helper instead? Talking about `IEnvironmentVariableHelper`. So that we can simplify tests.

Reply:
Fixed.